azkaban-exec-server/src/main/java/azkaban/execapp/reporter/AzkabanKafkaAvroEventReporter.java

```diff
@@ -0,0 +1,121 @@
+package azkaban.execapp.reporter;
+
+import azkaban.event.Event;
+import azkaban.utils.Props;
+import com.google.common.base.Preconditions;
+import gobblin.metrics.MetricContext;
+import gobblin.metrics.event.EventSubmitter;
+import gobblin.metrics.kafka.KafkaAvroEventReporter;
+import gobblin.metrics.kafka.KafkaAvroSchemaRegistry;
+import gobblin.metrics.kafka.KafkaEventReporter;
+import gobblin.metrics.kafka.KafkaPusher;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.log4j.Logger;
+
+
+/**
+ * Default implementation of the <code>AzkabanEventReporter</code> class.
+ * An instance of this class is injected via guice into the <code>FlowRunnerManager</code>
+ * class when event reporting is enabled.
+ */
+public class AzkabanKafkaAvroEventReporter implements AzkabanEventReporter {
+
+  private static final Logger logger = Logger.getLogger(AzkabanKafkaAvroEventReporter.class);
+  private static final String EVENT_REPORTING_KAFKA_BROKERS = "event.reporting.kafka.brokers";
+  private static final String EVENT_REPORTING_KAFKA_TOPIC = "event.reporting.kafka.topic";
+  private static final String EVENT_REPORTING_KAFKA_SCHEMA_REGISTRY_URL = "event.reporting.kafka.schema.registry.url";
+  private static final String ROOT_CONTEXT_NAME = "AzkabanKafkaEvents";
+  private static final String AZKABAN_FLOW_EVENTS_NAMESPACE = "AzkabanFlowEvents";
+  private static final String AZKABAN_JOB_EVENTS_NAMESPACE = "AzkabanJobEvents";
+  private final Properties kafkaProps = new Properties();
+  private MetricContext rootContext;
+  private KafkaAvroEventReporter kafkaAvroEventReporter;
+
+  // For testing only.
+  public AzkabanKafkaAvroEventReporter(final KafkaAvroEventReporter kafkaAvroEventReporter,
+      final Props props) {
+    this.kafkaAvroEventReporter = kafkaAvroEventReporter;
+    initKafkaProperties(props);
+  }
+
+  // Constructed via guice provider
+  public AzkabanKafkaAvroEventReporter(final Props props) {
+
+    initKafkaProperties(props);
+
+    final String eventReportingKafkaBrokers =
+        props.get(EVENT_REPORTING_KAFKA_BROKERS);
+    final String eventReportingKafkaTopic =
+        props.get(EVENT_REPORTING_KAFKA_TOPIC);
+
+    this.rootContext = MetricContext.builder(ROOT_CONTEXT_NAME).build();
+    final KafkaEventReporter.Builder<? extends KafkaAvroEventReporter.Builder> builder =
+        KafkaAvroEventReporter.Factory.forContext(this.rootContext);
+    final KafkaPusher kafkaPusher =
+        new AzkabanKafkaPusher(eventReportingKafkaBrokers, eventReportingKafkaTopic);
+
+    try {
+      this.kafkaAvroEventReporter = builder.withKafkaPusher(kafkaPusher)
+          .withSchemaRegistry(new KafkaAvroSchemaRegistry(this.kafkaProps))
+          .build(eventReportingKafkaBrokers, eventReportingKafkaTopic);
+      logger.info("Initialized a new kafkaAvroEventReporter");
+    } catch (final Throwable t) {
```

Comment:
Is catching RuntimeException enough?

Reply:
Yea, that should be enough. Will change...

Reply:
done.